### PR TITLE
haskellPackages: unbreak amazonka

### PR DESCRIPTION
**Pull request by jbgi**
and other packages broken due to amazonka.

### CODE REVIEW

**Comment by jbgi** [comment]
@peti should I run https://github.com/NixOS/cabal2nix/blob/master/update-nixpkgs.sh as part of this PR?

**Comment by peti** [comment]
No, that's not necessary. But you should target the branch haskell-updates.

Check out https://m.youtube.com/watch?v=KLhkAEk8I20 for detailed instructions.

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -1228,4 +1228,10 @@ self: super: {
     '';
   });
 
+  # https://github.com/brendanhay/amazonka/commit/657b70d174fe5cb61e56cb8b9c5e57f1ec216f2b
+  amazonka = appendPatch super.amazonka ./patches/amazonka-Allow-http-client-0.6.patch;
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-common.nix]: @jbgi Generally, people don't like to carry around patches in nixpkgs, but instead pull them from upstream.  

Could you change this to pull the patch from upstream, similar to below:

https://github.com/NixOS/nixpkgs/pull/68935#discussion_r324888857

Comment (jbgi) [reply pkgs/development/haskell-modules/configuration-common.nix]: I see. Though in this case there is a single [commit patch](https://github.com/brendanhay/amazonka/commit/657b70d174fe5cb61e56cb8b9c5e57f1ec216f2b.patch) that need to be split to apply to each package. Is there a precedent in nixpkgs of a similar situation?

Comment (cdepillabout) [reply pkgs/development/haskell-modules/configuration-common.nix]: @jbgi Ah, I see.  I don't know what to do in that situation.  I'd say it looks fine as-is.

**Review by cdepillabout — APPROVED** [review]
This looks good to me, although I didn't actually try building all these packages that have been marked unbroken.

cc @peti

**Comment by jbgi** [comment]
I tested the commit on top of `nixos-19.09` using
`nix-build -E 'with (import ./. { config = {allowBroken = true;}; }); haskellPackages.ghcWithPackages (p: with p; [amazonka-alexa-business amazonka-apigateway amazonka-application-autoscaling amazonka-appstream amazonka-appsync amazonka-athena amazonka-autoscaling amazonka-autoscaling-plans amazonka-batch amazonka-budgets amazonka-certificatemanager amazonka-certificatemanager-pca amazonka-cloud9 amazonka-clouddirectory amazonka-cloudformation amazonka-cloudfront amazonka-cloudhsm amazonka-cloudhsmv2 amazonka-cloudsearch amazonka-cloudsearch-domains amazonka-cloudtrail amazonka-cloudwatch amazonka-cloudwatch-events amazonka-cloudwatch-logs amazonka-codebuild amazonka-codecommit amazonka-codedeploy amazonka-codepipeline amazonka-codestar amazonka-cognito-identity amazonka-cognito-idp amazonka-cognito-sync amazonka-comprehend amazonka-config amazonka-connect amazonka-cost-explorer amazonka-cur amazonka-datapipeline amazonka-devicefarm amazonka-directconnect amazonka-discovery amazonka-dms amazonka-ds amazonka-dynamodb amazonka-dynamodb-dax amazonka-dynamodb-streams amazonka-ec2 amazonka-ecr amazonka-ecs amazonka-efs amazonka-elasticache amazonka-elasticbeanstalk amazonka-elasticsearch amazonka-elastictranscoder amazonka-elb amazonka-elbv2 amazonka-emr amazonka-fms amazonka-gamelift amazonka-glacier amazonka-glue amazonka-greengrass amazonka-guardduty amazonka-health amazonka-iam amazonka-importexport amazonka-inspector amazonka-iot amazonka-iot-analytics amazonka-iot-dataplane amazonka-iot-jobs-dataplane amazonka-kinesis amazonka-kinesis-analytics amazonka-kinesis-firehose amazonka-kinesis-video amazonka-kinesis-video-archived-media amazonka-kinesis-video-media amazonka-kms amazonka-lambda amazonka-lex-models amazonka-lex-runtime amazonka-lightsail amazonka-marketplace-analytics amazonka-marketplace-entitlement amazonka-marketplace-metering amazonka-mechanicalturk amazonka-mediaconvert amazonka-medialive amazonka-mediapackage amazonka-mediastore amazonka-mediastore-dataplane amazonka-migrationhub amazonka-ml amazonka-mobile amazonka-mq amazonka-opsworks amazonka-opsworks-cm amazonka-organizations amazonka-pinpoint amazonka-polly amazonka-pricing amazonka-rds amazonka-redshift amazonka-rekognition amazonka-resourcegroups amazonka-resourcegroupstagging amazonka-route53 amazonka-route53-autonaming amazonka-route53-domains amazonka-s3-streaming amazonka-sagemaker amazonka-sagemaker-runtime amazonka-sdb amazonka-secretsmanager amazonka-serverlessrepo amazonka-servicecatalog amazonka-ses amazonka-shield amazonka-sms amazonka-snowball amazonka-sns amazonka-sqs amazonka-ssm amazonka-stepfunctions amazonka-storagegateway amazonka-sts amazonka-support amazonka-swf amazonka-transcribe amazonka-translate amazonka-waf amazonka-waf-regional amazonka-workdocs amazonka-workmail amazonka-workspaces amazonka-xray aws-easy aws-ses-easy confcrypt ec2-unikernel ekg-cloudwatch heavy-logger-amazon serverless-haskell])'`

**Comment by cdepillabout** [comment]
@peti Could you cherry pick the commit from this PR (975941c) so that @jbgi doesn't have to keep rebasing this PR on top of `haskell-updates`?

**Comment by peti** [comment]
Cherry-picked as 982e840c12a4bbeccd621a05f80af590a2ea4324.

**Pull request closed without merging**